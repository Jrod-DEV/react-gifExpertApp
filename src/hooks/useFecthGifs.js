import { useEffect } from 'react';
import { useState } from 'react/cjs/react.development';
import { getGifs } from '../helpers/getGifs';

export const useFetchGifs = (category) => {
  const [state, setState] = useState({
    data: [],
    loading: true,
  });

  useEffect(() => {
    getGifs(category).then((imgs) => {
      setTimeout(() => {
        setState({
          data: imgs,
          loading: false,
        });
      }, 1500);
    });
  }, [category]);

  return state;
};
