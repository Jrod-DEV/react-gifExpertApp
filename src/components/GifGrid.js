import React from 'react';
import { useFetchGifs } from '../hooks/useFecthGifs';


export const GifGrid = ({ category }) => {

  const { loading } = useFetchGifs();

  return (
    <>
      <h3> {category} </h3>
      {loading ? 'Loading...' : 'Data charged!'}
{/* 
      {      <div className="card-grid">
        {images.map((img) => (
          <GifGridItem key={img.id} {...img} />
        ))}
      </div>}
       */}
    </>
  );
};
