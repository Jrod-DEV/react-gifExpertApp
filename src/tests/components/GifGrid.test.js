import React from 'react';
import '@testing-library/jest-dom';

import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFecthGifs';
jest.mock('../../hooks/useFecthGifs');

describe('Pruebas en <GifGrid/>', () => {
  const category = 'DBZ';

  test('match con snapshot', () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });

    const wrapper = shallow(<GifGrid category={category} />);

    expect(wrapper).toMatchSnapshot();
  });

  test('debe de mostrar items cuando se cargan imagenes useFetchGifs', () => {
    const gifs = [
      {
        id: 'ABC',
        url: 'http://example/cosa.jpg',
        title: 'Some stuff',
      },
      {
        id: 'ABC',
        url: 'http://example/cosa.jpg',
        title: 'Some stuff',
      },
    ];

    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false,
    });

    const wrapper = shallow(<GifGrid category={category} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('p').exists()).toBe(false);
    expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
  });

});
