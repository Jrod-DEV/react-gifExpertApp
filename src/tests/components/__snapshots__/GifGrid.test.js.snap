// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid/> debe de mostrar items cuando se cargan imagenes useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    DBZ
     
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Some stuff"
      url="http://example/cosa.jpg"
    />
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Some stuff"
      url="http://example/cosa.jpg"
    />
  </div>
</Fragment>
`;

exports[`Pruebas en <GifGrid/> match con snapshot 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    DBZ
     
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading your GIFS!
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
